transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java

```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
         assertTrue(future.isCancelled());
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void testWrongPromiseChannel() throws Exception {
+        ChannelPipeline pipeline = new LocalChannel().pipeline();
+        group.register(pipeline.channel());
```

Comment:
@normanmaurer hm, I copied this from your test and it doesn't have sync in it :).